engine/runtime/src/test/java/org/enso/compiler/test/context/JacksonTest.java

```diff
@@ -29,4 +30,27 @@ public void testSerdeOfSuggestion() throws Exception {
     assertEquals("doc", suggestion.documentation().get());
     assertEquals(Suggestion.Module.class, suggestion.getClass());
   }
+
+  @Test
+  public void testArraySerdeOfSuggestion() throws Exception {
+    Object shape = new Suggestion[]{new Suggestion.Module(
```

Comment:
Now I finally understand how Jackson works! It is not possible to serialize `Object[]`, but when one uses `Suggestion[]`, then everything works. Jackson doesn't inspect the actual values - but it inspects signatures - moreover `List<Suggestion>` isn't going to work due to type erasure.